scene/animation/animation_tree.h

```diff
@@ -416,9 +456,28 @@ class AnimationTree : public Node {
 	Vector3 get_root_motion_scale_accumulator() const;
 
 	real_t get_connection_activity(const StringName &p_path, int p_connection) const;
-	void advance(double p_time);
+	void advance(double p_time, bool p_record = false);
```

Comment:
Probably this optional argument is not necessary. If the default is false, you can simply fire save_record() immediately after advance() if you want it to be true.

Reply:
For consistence, I remove the optional argument of `load_record()`, if need to discard newest records, call `discard_latest_records()` manually.

Reply:
They could be implemented separately. Refer to erase() and pop_back() of the cpp array.